test/sparse/sparse.jl

```diff
@@ -646,321 +655,318 @@ end
-        a[:,2] = 2
-        @test countnz(a) == 19
-        @test a[:,2] == 2*sparse(ones(Int,10))
-        b = copy(a)
-
-        # Zero-assignment behavior of setindex!(A, v, i, j)
-        a[1,3] = 0
-        @test nnz(a) == 19
-        @test countnz(a) == 18
-        a[2,1] = 0
-        @test nnz(a) == 19
-        @test countnz(a) == 18
-
-        # Zero-assignment behavior of setindex!(A, v, I, J)
-        a[1,:] = 0
-        @test nnz(a) == 19
-        @test countnz(a) == 9
-        a[2,:] = 0
-        @test nnz(a) == 19
-        @test countnz(a) == 8
-        a[:,1] = 0
-        @test nnz(a) == 19
-        @test countnz(a) == 8
-        a[:,2] = 0
-        @test nnz(a) == 19
-        @test countnz(a) == 0
-        a = copy(b)
-        a[:,:] = 0
-        @test nnz(a) == 19
-        @test countnz(a) == 0
-
-        # Zero-assignment behavior of setindex!(A, B::SparseMatrixCSC, I, J)
-        a = copy(b)
-        a[1:2,:] = spzeros(2, 10)
-        @test nnz(a) == 19
-        @test countnz(a) == 8
-        a[1:2,1:3] = sparse([1 0 1; 0 0 1])
-        @test nnz(a) == 20
-        @test countnz(a) == 11
-        a = copy(b)
-        a[1:2,:] = let c = sparse(ones(2,10)); fill!(c.nzval, 0); c; end
-        @test nnz(a) == 19
-        @test countnz(a) == 8
-        a[1:2,1:3] = let c = sparse(ones(2,3)); c[1,2] = c[2,1] = c[2,2] = 0; c; end
-        @test nnz(a) == 20
-        @test countnz(a) == 11
-
-        a[1,:] = 1:10
-        @test a[1,:] == sparse([1:10;])
-        a[:,2] = 1:10
-        @test a[:,2] == sparse([1:10;])
-
-        a[1,1:0] = []
-        @test a[1,:] == sparse([1; 1; 3:10])
-        a[1:0,2] = []
-        @test a[:,2] == sparse([1:10;])
-        a[1,1:0] = 0
-        @test a[1,:] == sparse([1; 1; 3:10])
-        a[1:0,2] = 0
-        @test a[:,2] == sparse([1:10;])
-        a[1,1:0] = 1
-        @test a[1,:] == sparse([1; 1; 3:10])
-        a[1:0,2] = 1
-        @test a[:,2] == sparse([1:10;])
-
-        @test_throws BoundsError a[:,11] = spzeros(10,1)
-        @test_throws BoundsError a[11,:] = spzeros(1,10)
-        @test_throws BoundsError a[:,-1] = spzeros(10,1)
-        @test_throws BoundsError a[-1,:] = spzeros(1,10)
-        @test_throws BoundsError a[0:9] = spzeros(1,10)
-        @test_throws BoundsError a[:,11] = 0
-        @test_throws BoundsError a[11,:] = 0
-        @test_throws BoundsError a[:,-1] = 0
-        @test_throws BoundsError a[-1,:] = 0
-        @test_throws BoundsError a[0:9] = 0
-        @test_throws BoundsError a[:,11] = 1
-        @test_throws BoundsError a[11,:] = 1
-        @test_throws BoundsError a[:,-1] = 1
-        @test_throws BoundsError a[-1,:] = 1
-        @test_throws BoundsError a[0:9] = 1
-
-        @test_throws DimensionMismatch a[1:2,1:2] = 1:3
-        @test_throws DimensionMismatch a[1:2,1] = 1:3
-        @test_throws DimensionMismatch a[1,1:2] = 1:3
-        @test_throws DimensionMismatch a[1:2] = 1:3
-    end
-
-    let A = spzeros(Int, 10, 20)
```

Comment:
Collapsing this sequence of `let` blocks in the `setindex` tests into a single block could impact semantics?